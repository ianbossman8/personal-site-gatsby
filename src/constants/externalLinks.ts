export const EXTERNAL_LINKS = {
  LT: 'https://linktr.ee/ianbossman',
  GH: 'https://github.com/ianbossman8',
  M: 'https://ianbossman.medium.com/'
}
