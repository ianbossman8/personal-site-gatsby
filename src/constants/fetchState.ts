export enum FETCH_STATE {
  SUCCESS = 'SUCCESS',
  FAIL = 'FAIL',
  FETCHING = 'FETCHING'
}
