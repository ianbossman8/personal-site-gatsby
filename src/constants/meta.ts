import { PATH } from './path'

export const PAGE_META = {
  INDEX: {
    title: 'Home',
    pathname: PATH.ROOT
  },
  BLOG: {
    description: "List of awesome Ian Bossman's blogs",
    title: 'Blogs',
    pathname: PATH.BLOGS
  },
  404: {
    description: 'Page Not Found',
    title: '404',
    pathname: PATH[404]
  }
}
