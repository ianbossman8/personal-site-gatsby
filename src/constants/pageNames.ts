export const PAGE_NAMES = {
  ABOUT: 'about',
  HOME: 'home',
  CONTACT: 'contact',
  BLOGS: 'blogs',
  PROJECTS: 'projects',
  PROMOTIONS: 'promotions',
  PRIVACY: 'privacy',
  404: '404'
}
