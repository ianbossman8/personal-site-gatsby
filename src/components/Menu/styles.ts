import styled, { css } from 'styled-components'
import { MEDIA_QUERY_MEDIUM_WIDTH_RULE } from '../../constants/styles'

export const MenuContainer = styled.div<{ showMenu: boolean }>`
  ${({ showMenu }) =>
    css`
      ${MEDIA_QUERY_MEDIUM_WIDTH_RULE} {
        position: fixed;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        background: ${({ theme: { colours } }) => colours.primary.blur.strong};
        -webkit-backdrop-filter: blur(0.0125rem);
        backdrop-filter: blur(0.0125rem);
        display: ${showMenu ? 'flex' : 'none'};
        justify-content: center;
        z-index: 1;
      }
    `}
`
