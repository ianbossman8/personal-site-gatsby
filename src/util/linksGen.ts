import { PageMeta } from '../constants/meta'
import { PAGE_NAMES } from '../constants/pageNames'
import symbols from '../constants/symbols'

export interface LinksObj {
  title: string
  icon: number
  pathname: string
}

const iconMapper = {
  [PAGE_NAMES.HOME]: symbols.house,
  [PAGE_NAMES.ABOUT]: symbols.monkey,
  [PAGE_NAMES.BLOGS]: symbols.writingHand,
  [PAGE_NAMES.PROJECTS]: symbols.tools,
  [PAGE_NAMES.PRIVACY]: symbols.detective,
  [PAGE_NAMES.PROMOTIONS]: symbols.gift
}

export default function linksGen(pageMeta: PageMeta, filter: string[]) {
  return Object.keys(pageMeta).reduce((acc, cur) => {
    const pathname = pageMeta[cur].pathname

    if (!filter.includes(pageMeta[cur].title)) {
      acc.push({
        title: pageMeta[cur].title,
        icon: iconMapper[pageMeta[cur].title],
        pathname
      })
    }

    return acc
  }, [] as LinksObj[])
}
