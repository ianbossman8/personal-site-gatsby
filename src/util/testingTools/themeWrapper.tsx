import React from 'react'
import { ThemeProvider } from 'styled-components'
import THEME from '../../constants/theme'
import { styleTheme } from '../../styles/theme'

export default function themeWrapper(component: React.ReactElement) {
  return <ThemeProvider theme={styleTheme[THEME.LIGHT]}>{component}</ThemeProvider>
}
