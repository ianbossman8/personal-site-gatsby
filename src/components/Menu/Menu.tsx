import React, { useEffect, useState } from 'react'
import { IoIosMenu } from 'react-icons/io'
import { GrClose } from 'react-icons/gr'
import LinksList from '../LinksList/LinksList'
import useModalEscapeKeyHooks from '../../customHooks/useModalEscapeKeyHooks'
import { LinksObj } from '../../util/linksGen'
import { SIZE } from '../../constants/font'
import { MenuContainer } from './styles'
import { MenuButton } from '../../styles/buttons'
import useWindowResizeHook from '../../customHooks/useWindowResizeHook'
import { WIDTH_BOUNDARIES } from '../../constants/styles'

interface Props {
  linksMeta: LinksObj[]
  size?: SIZE
  isIndex: boolean
}

export default function Menu(props: Props) {
  const { size, isIndex, linksMeta } = props
  const curWidth = useWindowResizeHook('width')
  const [showMenu, setShowMenu] = useState<boolean>(false)

  function handleMenuClick() {
    setShowMenu((showMenu) => !showMenu)
  }

  function handleMenuClose() {
    setShowMenu(false)
  }

  useModalEscapeKeyHooks(handleMenuClose)

  useEffect(() => {
    if (curWidth && curWidth >= WIDTH_BOUNDARIES.M) {
      setShowMenu(false)
    }
  }, [curWidth])

  return (
    <>
      {!isIndex && (
        <>
          <MenuContainer showMenu={showMenu} onClick={handleMenuClick}>
            <LinksList linksMeta={linksMeta} size={size} />
          </MenuContainer>
          <MenuButton>
            {!showMenu ? (
              <IoIosMenu size={48} onClick={handleMenuClick} title="menu" />
            ) : (
              <GrClose size={42} onClick={handleMenuClick} title="close" />
            )}
          </MenuButton>
        </>
      )}
    </>
  )
}
