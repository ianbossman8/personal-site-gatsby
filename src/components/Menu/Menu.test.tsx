import React from 'react'
import renderer from 'react-test-renderer'
import Menu from './Menu'
import themeWrapper from '../../util/testingTools/themeWrapper'
import { pageMeta } from '../../constants/meta'
import linksGen from '../../util/linksGen'
import { PAGE_NAMES } from '../../constants/pageNames'

describe('Menu', () => {
  describe('snapshot', () => {
    it('should render the menu', () => {
      const linksMeta = linksGen(pageMeta, [PAGE_NAMES[404], PAGE_NAMES.PRIVACY])
      const tree = renderer
        .create(themeWrapper(<Menu linksMeta={linksMeta} isIndex={false} />))
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should not render the menu', () => {
      const linksMeta = linksGen(pageMeta, [PAGE_NAMES[404], PAGE_NAMES.PRIVACY])
      const tree = renderer
        .create(themeWrapper(<Menu linksMeta={linksMeta} isIndex={true} />))
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
