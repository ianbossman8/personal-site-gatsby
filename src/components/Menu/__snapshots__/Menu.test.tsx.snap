// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshot should not render the menu 1`] = `null`;

exports[`Menu snapshot should render the menu 1`] = `
Array [
  <div
    className="sc-fujyAs jbPEWQ"
    onClick={[Function]}
  >
    <ul
      className="sc-iCoGMd kzyMyM"
      direction="row"
      fontSize="N"
    >
      <li>
        <span
          aria-label="Home"
          className="sc-bdnxRM hHhLGM"
          role="img"
          size="N"
        >
          🏡
        </span>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <span
          aria-label="About"
          className="sc-bdnxRM hHhLGM"
          role="img"
          size="N"
        >
          🐵
        </span>
        <a
          href="/about/"
        >
          About
        </a>
      </li>
      <li>
        <span
          aria-label="Blogs"
          className="sc-bdnxRM hHhLGM"
          role="img"
          size="N"
        >
          ✍
        </span>
        <a
          href="/blogs/"
        >
          Blogs
        </a>
      </li>
      <li>
        <span
          aria-label="Projects"
          className="sc-bdnxRM hHhLGM"
          role="img"
          size="N"
        >
          🔨
        </span>
        <a
          href="/projects/"
        >
          Projects
        </a>
      </li>
      <li>
        <span
          aria-label="Promotions"
          className="sc-bdnxRM hHhLGM"
          role="img"
          size="N"
        >
          🎁
        </span>
        <a
          href="/promotions/"
        >
          Promotions
        </a>
      </li>
    </ul>
  </div>,
  <button
    className="sc-kEqXSa gmvYaa"
  >
    <svg
      fill="currentColor"
      height={48}
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width={48}
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        menu
      </title>
      <path
        d="M432 176H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 272H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 368H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16z"
      />
    </svg>
  </button>,
]
`;
